cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/DatasetServiceClient.java

```diff
@@ -82,13 +83,16 @@ public EndpointStrategy get() {
   }
 
   @Nullable
-  public DatasetMeta getInstance(String instanceName, @Nullable Id owner) throws DatasetManagementException {
-    String query = "";
-    if (owner != null) {
-      query = "?ownerType=" + owner.getIdType() + "&ownerId=" + owner;
+  public DatasetMeta getInstance(String instanceName,
+                                 @Nullable List<? extends Id> owners) throws DatasetManagementException {
+    Multimap<String, String> headers = HashMultimap.create();
+    if (owners != null) {
+      for (Id owner : owners) {
+        headers.put(Constants.Header.DATASET_OWNER, owner.getIdType() + "//" + owner.getIdRep());
```

Comment:
`Joiner` for two elements (`owner.getIdType()` and `owner.getIdRep()`)? Not sure what you mean.